s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/SceneRepositoryImpl.java

```diff
@@ -0,0 +1,352 @@
+import pl.cyfronet.s4e.properties.DbProperties;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class SceneRepositoryImpl implements SceneRepositoryExt {
+    private final DbProperties dbProperties;
+    private Connection conn;
+    private int paramIndex;
+    private final JdbcTemplate jdbcTemplate;
+
+    public SceneRepositoryImpl(DbProperties dbProperties, JdbcTemplate jdbcTemplate) throws SQLException {
+        this.dbProperties = dbProperties;
+        this.conn = DriverManager.getConnection(dbProperties.getUrl(),
+                                                dbProperties.getUsername(),
+                                                dbProperties.getPassword());
+        this.paramIndex = 1;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    public List<Scene> findAllByParamsMap(Map<String, Object> params) throws SQLException {
+        return jdbcTemplate.query(prepareQuery(params), (rs, rowNum) -> Scene.builder()
+                .id(rs.getLong("id"))
+                .product(Product.builder()
+                        .id(rs.getLong("product_id"))
+                        .build())
+                .build());
+    }
+
+    private String prepareQuery(Map<String, Object> params) throws SQLException {
+        PreparedStatement ps = conn.prepareStatement(prepareSqlQuery(params));
+        paramIndex = 1;
+        prepareParameters(ps, params);
+        log.trace(ps.toString());
+        return ps.toString();
+    }
+
+    private String prepareSqlQuery(Map<String, Object> params) {
+        StringBuilder resultQuery = new StringBuilder();
+        resultQuery.append("SELECT id, product_id FROM Scene WHERE true ");
+
+        queryByTime(params, resultQuery);
+        queryByText(params, resultQuery);
+        queryByNumber(params, resultQuery);
+        // [] inputAuxiliaryFiles -> ...
+        // [] polygon -> field[footprint]
+        if (params.containsKey("orderBy") && params.containsKey("sortBy")) {
+            resultQuery.append(" ORDER BY " + getOrderByField(params) + " " + getOrderBy(params));
+        }
+        resultQuery.append(" LIMIT ? ");
+        resultQuery.append(" OFFSET ? ");
+        return resultQuery.append(";").toString();
+    }
+
+    private void prepareParameters(PreparedStatement ps, Map<String, Object> params) throws SQLException {
+        timeParameters(ps, params);
+        textParameters(ps, params);
+        numberParameters(ps, params);
+        ps.setInt(paramIndex++, Integer.parseInt(params.getOrDefault("limit", 25).toString()));
+        ps.setInt(paramIndex++, Integer.parseInt(params.getOrDefault("offset", 0).toString()));
+
+    }
+
+    private String getOrderByField(Map<String, Object> params) {
+        switch (params.get("sortBy").toString()) {
+            case "ingestion_time":
+                return "ingestion_time";
```

Comment:
`ingestion_time` and other switch cases should be stored as constants in SQL Model.